### ISSUE TITLE
False positive Unable to import with standard modules like os , logging and python embedded, inside the workspace

### ISSUE THREAD

**Issue by angelmoralesruiz** [pylint-dev/pylint#10607]
### Bug description

```python
False positive Unable to import error of standar/included modules with python. (os, logging, datetime, etc.) 

Using python 3.13.17 embedded with pylint module installed. Python is part of the workspace.
```

### Configuration

```ini
"python.testing.pytestEnabled": true,
		"python.testing.unittestEnabled": false,
		"python.testing.pytestPath": "${workspaceFolder}\\3rd_bin\\python\\Scripts\\pytest.exe",
		"editor.linkedEditing": true,
		"python.analysis.autoImportCompletions": true,
		"python.analysis.exclude": [
			"3rd_bin"
		],
		"python.analysis.diagnosticMode": "workspace",
		"pylint.args": [
			"--max-line-length=100",
			"--init-hook=import sys; sys.path.append('.')"
		],
		"pylint.path": [
			"${workspaceFolder}\\3rd_bin\\python\\Scripts\\pylint.exe"
		],
		"pylint.interpreter": [
			"${workspaceFolder}\\3rd_bin\\python\\python.exe"
		],
		"pylint.importStrategy": "fromEnvironment",
```

### Command used

```shell
Using pylint installed in python embedded from Pylint extension

Identifier ms-python.pylint
Version 2025.2.0
```

### Pylint output

```python
Unable to import 'logging'PylintE0401:import-error
Unable to import 'os'PylintE0401:import-error

2025-10-03 09:38:26.005 [info] Name: Pylint
2025-10-03 09:38:26.006 [info] Module: pylint
2025-10-03 09:38:26.006 [info] Python extension loading
2025-10-03 09:38:26.006 [info] Waiting for interpreter from Python extension.
2025-10-03 09:38:31.048 [info] Interpreter from setting pylint.interpreter: ${workspaceFolder}\3rd_bin\python\python.exe
2025-10-03 09:38:31.053 [info] Interpreter from setting pylint.interpreter: ${workspaceFolder}\3rd_bin\python\python.exe
2025-10-03 09:38:31.053 [info] Python extension loaded
2025-10-03 09:38:31.975 [info] Server run command: c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\python.exe c:\Users\ut\.vscode\extensions\ms-python.pylint-2025.2.0\bundled\tool\lsp_server.py
2025-10-03 09:38:31.981 [info] Server: Start requested.
2025-10-03 09:38:42.780 [info] CWD Server: c:\Users\ut\VSCodeProjects\prau
2025-10-03 09:38:42.782 [info] Settings used to run Server:
[
    {
        "enabled": true,
        "cwd": "c:\\Users\\ut\\VSCodeProjects\\prau",
        "workspace": "file:///c%3A/Users/ut/VSCodeProjects/prau",
        "args": [
            "--max-line-length=100",
            "--init-hook=import sys; sys.path.append('.')"
        ],
        "severity": {
            "convention": "Information",
            "error": "Error",
            "fatal": "Error",
            "refactor": "Hint",
            "warning": "Warning",
            "info": "Information"
        },
        "path": [
            "c:\\Users\\ut\\VSCodeProjects\\prau\\3rd_bin\\python\\Scripts\\pylint.exe"
        ],
        "ignorePatterns": [],
        "interpreter": [
            "c:\\Users\\ut\\VSCodeProjects\\prau\\3rd_bin\\python\\python.exe"
        ],
        "importStrategy": "fromEnvironment",
        "showNotifications": "onWarning",
        "extraPaths": []
    }
]

2025-10-03 09:38:42.782 [info] Global settings:
{
    "cwd": "${workspaceFolder}",
    "enabled": true,
    "workspace": "C:\\Users\\ut\\AppData\\Local\\Programs\\Microsoft VS Code",
    "args": [],
    "severity": {
        "convention": "Information",
        "error": "Error",
        "fatal": "Error",
        "refactor": "Hint",
        "warning": "Warning",
        "info": "Information"
    },
    "path": [],
    "ignorePatterns": [],
    "interpreter": [],
    "importStrategy": "useBundled",
    "showNotifications": "onWarning",
    "extraPaths": []
}

2025-10-03 09:38:42.782 [info] sys.path used to run Server:
   c:\Users\ut\.vscode\extensions\ms-python.pylint-2025.2.0\bundled\tool
   c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\python313.zip
   c:\Users\ut\VSCodeProjects\prau\3rd_bin\python
   c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\Lib\site-packages
   c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\Lib\site-packages\win32
   c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\Lib\site-packages\win32\lib
   c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\Lib\site-packages\Pythonwin
   
   C:\Users\ut\.vscode\extensions\ms-python.python-2025.14.0-win32-x64\python_files
   c:\Users\ut\.vscode\extensions\ms-python.pylint-2025.2.0\bundled\libs
   c:\Users\ut\VSCodeProjects\prau
2025-10-03 09:38:42.892 [info] c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\Scripts\pylint.exe --version
2025-10-03 09:38:42.892 [info] CWD Server: c:\Users\ut\VSCodeProjects\prau
2025-10-03 09:39:07.733 [info] 
pylint 3.3.8
astroid 3.3.11
Python 3.13.7 (tags/v3.13.7:bcee1c3, Aug 14 2025, 14:15:11) [MSC v.1944 64 bit (AMD64)]


2025-10-03 09:39:07.767 [info] Version info for linter running for C:\Users\ut\VSCodeProjects\prau:
pylint 3.3.8
astroid 3.3.11
Python 3.13.7 (tags/v3.13.7:bcee1c3, Aug 14 2025, 14:15:11) [MSC v.1944 64 bit (AMD64)]

2025-10-03 09:39:07.774 [info] SUPPORTED pylint>=2.12.2
FOUND pylint==3.3.8

2025-10-03 09:39:07.888 [info] [Trace - 9:39:07 AM] Sending request 'textDocument/codeAction - (1)'.
2025-10-03 09:39:08.077 [info] c:\Users\ut\.vscode\extensions\ms-python.pylint-2025.2.0\bundled\tool\lsp_server.py:103: DeprecationWarning: 'workspace.get_document' has been deprecated, use 'workspace.get_text_document' instead
  document = LSP_SERVER.workspace.get_document(params.text_document.uri)

2025-10-03 09:39:08.087 [info] [Trace - 9:39:08 AM] Received notification 'window/logMessage'.
2025-10-03 09:39:08.087 [info] c:\Users\ut\VSCodeProjects\prau\3rd_bin\python\Scripts\pylint.exe --reports=n --output-format=json --max-line-length=100 --init-hook=import sys; sys.path.append('.') --clear-cache-post-run=y --from-stdin C:\Users\ut\VSCodeProjects\prau\base_lib\base_actions.py
2025-10-03 09:39:08.087 [info] [Trace - 9:39:08 AM] Received notification 'window/logMessage'.
2025-10-03 09:39:08.087 [info] CWD Server: c:\Users\ut\VSCodeProjects\prau
2025-10-03 09:39:08.414 [info] [Trace - 9:39:08 AM] Sending request 'textDocument/codeAction - (2)'.
2025-10-03 09:39:08.417 [info] [Trace - 9:39:08 AM] Sending notification '$/cancelRequest'.
2025-10-03 09:39:13.422 [info] [Trace - 9:39:13 AM] Sending notification '$/cancelRequest'.
2025-10-03 09:39:13.425 [info] [Trace - 9:39:13 AM] Sending request 'textDocument/codeAction - (3)'.
2025-10-03 09:39:39.117 [info] [Trace - 9:39:39 AM] Received notification 'window/logMessage'.
2025-10-03 09:39:39.117 [info] file:///c%3A/Users/ut/VSCodeProjects/prau/base_lib/base_actions.py :
```

### Expected behavior

Not show error Unable to import of a working import

### Pylint version

```shell
pylint 3.3.8
astroid 3.3.11
Python 3.13.7 (tags/v3.13.7:bcee1c3, Aug 14 2025, 14:15:11) [MSC v.1944 64 bit (AMD64)]
```

### OS / Environment

Windows 11 Enterprise 22H2  64 bits

### Additional dependencies

```python

```

**Comment by DanielNoord** [comment]
Have you opened this before in the extension repository? It might be related to the extension and not `pylint` itself

**Comment by Pierre-Sassoulas** [comment]
I'm almost certain this is either an issue in the extension as Daniel said (its configuration probably), or an environnement issue.

**Comment by angelmoralesruiz** [comment]
No. Sorry for the missunderstanding I will opne it.
